### PR TITLE
Add missing json tags to extender config

### LINKED ISSUE #65414: scheduler: fix API inconsistencies and test compatibility
/kind bug
/sig scheduling

There are many APIs which have been excluded in https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithmprovider/defaults/compatibility_test.go. For instance, `ExtenderConfig` is absent (found during https://github.com/kubernetes/kubernetes/issues/65413).

The tests should include all API types to ensure backward compatibility and API consistency. 

/cc @liggitt

### PR DESCRIPTION
**Pull request by ravigadde**
**What this PR does / why we need it**:
BindVerb was missing a json tag and so did TLSConfig.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #65414 

**Special notes for your reviewer**:
Please check and let me know whether this will address the compatibility issues.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by ravigadde** [comment]
@liggitt Is this a known issue?

```
W0720 21:05:05.088] W0720 21:05:05.088056    5836 executor.go:130] error running task "VPC/e2e-98012-dba53.test-cncf-aws.k8s.io" (9m58s remaining to succeed): error creating VPC: VpcLimitExceeded: The maximum number of VPCs has been reached.
```

**Comment by ravigadde** [comment]
Never mind. This seems to be issue: #66428

**Review by resouer — APPROVED** [review]
/lgtm

The CI failure seems to be unrelated.

**Comment by nikhita** [comment]
/retest

/assign @k82cn 
for approval

**Comment by nikhita** [comment]
@ravigadde Looks like this is ready to go, but just needs a rebase. :/

**Comment by ravigadde** [comment]
@liggitt Could you please lgtm this PR?

File: pkg/scheduler/api/v1/types.go [hunk 1]
```diff
@@ -169,11 +195,11 @@ type ExtenderConfig struct {
 	// Verb for the bind call, empty if not supported. This verb is appended to the URLPrefix when issuing the bind call to extender.
 	// If this method is implemented by the extender, it is the extender's responsibility to bind the pod to apiserver. Only one extender
 	// can implement this function.
-	BindVerb string
+	BindVerb string `json:"bindVerb,omitempty"`
```
Comment (liggitt) [inline pkg/scheduler/api/v1/types.go]: for posterity, this remains compatible because a custom unmarshaler was implemented to remain case insensitive

**Pull request merged**